src/plots/plots.js

```diff
@@ -659,6 +659,41 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
     }
 };
 
+// This function clears defaults between the first and second pass of
+// supplyDefaults. It exists because otherwise null attributes are
+// supplyDefault'd and inherited as *colors* instead of an actual null
+// attribute which needs to be supplydefaulted by the individual
+// expanded traces.
+plots.clearExpandedTraceDefaultColors = function(expandedTraces) {
+    var colorAttrs, path, trace, i, j;
+
+    // A better check *might* be to explicitly check for a groupby transform
```

Comment:
It's a hypothetical for now, since `groupby` is currently the only one that can yield more than one trace... but my rationale for suggesting this was that it's precisely this condition that breaks default colors. We'll see what happens if/when we add more such transforms, if this is insufficient then a category system is a great idea!